### PR TITLE
rgw_usage:CORTX-30360: Added Storage Stats as part of usage API to get capacity used by a user.

### PR DESCRIPTION
**Pull request by Diwakar92**
Problem : Storage stats for a user was not there as part of usage API.
Solution : Updated stats for a user and added those fields as part of usage API.
Signed-off-by: Diwakar92 <diwakar.kumar@seagate.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_usage.cc [hunk 1]
```diff
@@ -146,6 +146,37 @@ int RGWUsage::show(const DoutPrefixProvider *dpp, rgw::sal::Store* store,
     formatter->close_section(); // summary
   }
 
+  ret = rgw_user_sync_all_stats(dpp, store, user, null_yield);
```
Comment (sachinpunadikar) [inline src/rgw/rgw_usage.cc:149]: Add comment on explaining why you have added code here & also what all things you are doing here.

**Review by bkirunge7 — APPROVED** [review]

File: src/rgw/rgw_usage.cc (hunk 1 above)
Comment (Diwakar92) [reply src/rgw/rgw_usage.cc:153]: Addressed

File: src/rgw/rgw_usage.cc [hunk 2]
```diff
@@ -145,6 +145,41 @@ int RGWUsage::show(const DoutPrefixProvider *dpp, rgw::sal::Store* store,
 
     formatter->close_section(); // summary
   }
+  
+  // Added storage ststs for a user as part of this API
```
Comment (sumedhak27) [inline src/rgw/rgw_usage.cc:149]: typo `ststs -> stats`

**Comment by sumedhak27** [comment]
suggested commit title:
`rgw_usage: [CORTX-30360] Add Storage Stats to Usage Info` or some better term for `Usage Info`

**Review by sumedhak27 — APPROVED** [review]

File: src/rgw/rgw_usage.cc (hunk 2 above)
Comment (Diwakar92) [reply src/rgw/rgw_usage.cc]: Addressed

**Review by sachinpunadikar — APPROVED** [review]
+1

**Comment by sumedhak27** [comment]
Commit msg guideline:
1. commit should not in the past tense, it should be **imperative**.
2. There should **not** be a **period** at the end of the subject line.
3. We should start with `rgw` or `rgw_sal_motr` or the correct file/class/component name of change.
4. We can also include the ticket number.

https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/939720719/Creating+RGW+Pull+Requests+-+Guideline

**Pull request merged**